consensus/spos/bls/benchmark_verify_signatures_test.go

```diff
@@ -0,0 +1,108 @@
+package bls_test
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-crypto-go/signing"
+	"github.com/multiversx/mx-chain-crypto-go/signing/mcl"
+	"github.com/stretchr/testify/require"
+
+	crypto2 "github.com/multiversx/mx-chain-crypto-go"
+	multisig2 "github.com/multiversx/mx-chain-crypto-go/signing/mcl/multisig"
+	"github.com/multiversx/mx-chain-crypto-go/signing/multisig"
+
+	"github.com/multiversx/mx-chain-go/consensus/spos/bls"
+	dataRetrieverMocks "github.com/multiversx/mx-chain-go/dataRetriever/mock"
+
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/factory/crypto"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	"github.com/multiversx/mx-chain-go/testscommon/consensus"
+	"github.com/multiversx/mx-chain-go/testscommon/cryptoMocks"
+	"github.com/multiversx/mx-chain-go/testscommon/enableEpochsHandlerMock"
+	"github.com/multiversx/mx-chain-go/testscommon/statusHandler"
+)
+
+// Function to make a predictable iteration on keys from a map of keys
+func createListFromMapKeys(mapKeys map[string]crypto2.PrivateKey) []string {
+	keys := make([]string, 0, len(mapKeys))
+
+	for key := range mapKeys {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
+
+	return keys
+}
+
+// Benchmark on measuring performance
+func BenchmarkSubroundEndRound_VerifyNodesOnAggSigFailTime(b *testing.B) {
+	b.ResetTimer()
+	b.StopTimer()
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
to be easier to read + less prone to errors, after this line add a :point_down: and remove the cancel call at the end of the test
```go
defer func() {
   cancel()
}
```